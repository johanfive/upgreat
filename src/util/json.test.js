const fs = require('./fs')
jest.mock('./fs')

const json = require('./json')

describe('readJSONFile', () => {
  it('should throw error if path is wrong', async () => {
    fs.readFile.mockImplementationOnce(() => Promise.reject())

    await expect(json.readJSONFile('fake.json')).rejects.toMatchSnapshot()
  })

  it('should throw error if json parse fails', async () => {
    fs.readFile.mockImplementationOnce(() => Promise.resolve('lol'))

    await expect(json.readJSONFile('notjson.txt')).rejects.toMatchSnapshot()
  })

  it('should return parsed json', async () => {
    fs.readFile.mockImplementationOnce(() => Promise.resolve('{"hello": true}'))

    await expect(json.readJSONFile('notjson.txt')).resolves.toMatchSnapshot()
  })
})
