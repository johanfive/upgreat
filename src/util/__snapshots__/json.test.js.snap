// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readJSONFile should return parsed json 1`] = `
Object {
  "hello": true,
}
`;

exports[`readJSONFile should throw error if json parse fails 1`] = `[Error: could not parse notjson.txt, is it a valid JSON file?]`;

exports[`readJSONFile should throw error if path is wrong 1`] = `[Error: could not read fake.json, is the repo path correct?]`;
